test/Microsoft.AspNet.StaticFiles.Tests/StaticFileContextTest.cs

```diff
@@ -49,13 +51,54 @@ public void LookupFileInfo_ReturnsTrue_IfFileExists()
             var context = new StaticFileContext(httpContext, options, pathString, NullLogger.Instance);
 
             // Act
-            context.ValidatePath();
+            context.MatchPath();
             var result = context.LookupFileInfo();
 
             // Assert
             Assert.True(result);
         }
 
+        [Fact]
+        public void ValidatePath_ReturnsFalse_IfPathContainsInvalidCharacters()
+        {
+            // Test whole set of invalid chars in one test as testing indivually may be slow due to number of invalid path chars on Windows
+            var invalidPath = new string(Path.GetInvalidPathChars());
+
+            // Arrange
+            var options = new StaticFileOptions();
+            options.FileProvider = new TestFileProvider();
+            var pathString = new PathString("/test");
+            var httpContext = new DefaultHttpContext();
+            httpContext.Request.Path = new PathString($"/test/{invalidPath}.txt");
+            var context = new StaticFileContext(httpContext, options, PathString.Empty, NullLogger.Instance);
+
+            // Act
+            context.MatchPath();
```

Comment:
I was trying to be consistent with some of the tests which don't assert on MatchPath, but just noticed some do. I've added the asserts to my unit tests as they can't hurt. Will see if the latest commits pass CI and then squash.